flink/flink1.10-shims/src/main/java/org/apache/zeppelin/flink/Flink110Shims.java

```diff
@@ -337,4 +344,27 @@ public Map extractTableConfigOptions() {
     }
     return fields;
   }
+
+  public boolean isTimeIndicatorType(Object type) {
+    return FlinkTypeFactory.isTimeIndicatorType((TypeInformation<?>) type);
+  }
+
+  @Override
+  public Object lookupExecutor(ClassLoader classLoader, Object environmentSettings, Object sEnv) {
```

Comment:
Every implementation of `isTimeIndicatorType()`, `lookupExecutor()` seems be the same. Is it better move it to `FlinkShims` class?

Reply:
flink-shims don't have flink dependencies, so can not use these flink classes. Only sub-modules has flink dependency. This do cause code duplication, but I don't have better way to avoid that.

Reply:
As bellow, the class `Flink110Shims` inherits the class `FlinkShims`.
Do you mean the method `Flink110Shims::lookupExecutor()` can't be moved to `FlinkShims::lookupExecutor()` due to a dependency issue?
```
public class Flink110Shims extends FlinkShims
```

Reply:
Yes, if you take a look at the pom.xml under `flink-shims` module, there's no flink dependency, because we can not specify flink dependency here, it leave its child module to decide which flink version to use.

Reply:
Thanks for the detailed explanation.